File: Java/Vko6/OpiskelijaSingleton.java
import java.util.ArrayList;

public class OpiskelijaSingleton {

    //Luokalle arraylist opiskelija olioista
    private ArrayList<Opiskelija> opiskelijat = new ArrayList<Opiskelija>();
    //instanssin alustus
    private static OpiskelijaSingleton op_instance;
    //instanssin luonti metodi
    public static OpiskelijaSingleton getInstance() {
        if (op_instance == null) {
            // Luodaan instanssi jos sitä ei ole olemassa
            op_instance = new OpiskelijaSingleton();
        }
        return op_instance;
    }

    //private rakentaja
    private OpiskelijaSingleton() {

    }
    public void lisaaOpiskelija(Opiskelija oppilas) {
        opiskelijat.add(oppilas);
        System.out.println("Opiskelija " + oppilas.getNimi() + " lisatty!");

    }
    public void haeOpiskelija(String aName) {
        for (Opiskelija o : opiskelijat) {
            String name = o.getNimi();
            System.out.println(name.equals(aName));
            if (name.equals(aName)) {
                System.out.println("Opiskelija " + name + " löytyi!\n");
                o.tulostaTiedot();
                break;
            }
        System.out.println("Ei opiskelijaa " + name + " listassa.");
        }
    }
    public void tulostaOpiskelijat() {
        for (Opiskelija o : opiskelijat) {
            o.tulostaTiedot();
        }
    }
}

File: C++/Vko6/OpiskelijaSingleton.cpp
#include "OpiskelijaSingleton.h"
#include "opiskelija.h"
#include <iostream>

using namespace std;

//Luokan instanssin alustus
OpiskelijaSingleton* OpiskelijaSingleton::instance = nullptr;
// Luokan tietojäseneksi tallenttu opiskelija olio vektori nimeltä oppilaat
vector<opiskelija*> oppilaat;

// Instanssin luomiseen metodi
OpiskelijaSingleton* OpiskelijaSingleton::getInstance() {

    if (instance == nullptr) {
    instance = new OpiskelijaSingleton();
    }

    return instance;
}
//Rakentaja
OpiskelijaSingleton::OpiskelijaSingleton() {
}
//Purkaja
OpiskelijaSingleton::~OpiskelijaSingleton() {
}

//Opiskelijan lisäys oppilaisiin, parametrina opiskelija olion osoitin
void OpiskelijaSingleton::lisaaOpiskelija(opiskelija* oppilas) {
    //Lisätään vektorin loppuun syötetty olio
    oppilaat.push_back(oppilas);

}
//Haetaan opiskelija nimen perusteella
void OpiskelijaSingleton::haeOpiskelija(string aNimi) {
    //Pyydetään jokaiselta oliolta nimi ja verrataan sitä attribuuttiin
    //Käydään läpi jokainen opiskelija olioon osoittava osoitin
    for (auto &oppilas : oppilaat) {
        string name = oppilas->getNimi();
        if (name == aNimi) {
            cout << "Oppilas loytyi! "<< endl;
            //Oppilaan löytyessä tulostetaan tiedot
            oppilas->tulostaTiedot();
        }
        else {
            cout << "Ei oppilasta nimelta " << aNimi << endl;
        }
    }
}
//Tulostetaan tiedot jokaisen oppilaat vektorin opiskelijan olion metodilla tulostaTiedot.
void OpiskelijaSingleton::tulostaOpiskelijat() {
    // auto = ei tarvitse itse määrittää tyyppiä, ( ei meinaan ollu "opiskelija")
    for (auto &oppilas : oppilaat) {
        oppilas->tulostaTiedot();
    }
}